src/include/miopen/rnn/base_ops.hpp

```diff
@@ -74,6 +100,7 @@ class RnnBaseFunctions
                                                miopenDataType_t data_type,
                                                bool add_assign = true)
     {
+#if MIOPEN_USE_GEMM && MIOPEN_BACKEND_HIP
```

Comment:
This is main fix for the issue. The rest is the decoration related to this method of distribution of dependencies.